File: Tugas Besar 1/Product.cpp
/**
 * @file Product.cpp
 * @author 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @brief file berisi definisi class Product
 * @version 0.1
 * @date 2019-03-20
 */

/* NIM/Nama  : 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * Nama file : Product.cpp
 * Topik     : Tugas Besar 1 IF2210 - Pemograman Berorientasi Objek
 * Tanggal   : 20 Maret 2019
 * Deskripsi : Implementasi Product.cpp */

#include "Product.hpp"
#include <iostream>
#include <string.h>
using namespace std;

/**
 * @brief Create the Farm Product object
 * 
 */
Product::Product(){
	price = 0;
	this->name = new char[10];
	strcpy(name,"product stub");
}

/**
 * @brief Create the Farm Product object
 * 
 */
Product::Product(int theprice, char* thename){
	price = theprice;
	this->name = new char[10];
	strcpy(name,thename);
}

/**
 * @brief Get the Name object
 * 
 * @return char* 
 */
char* Product::getName(){
	return name;
}

/**
 * @brief Get the Price object
 * 
 * @return int 
 */
int Product::getPrice(){
	return price;
}

/**
 * @brief Set the Name object
 * 
 */
void Product::setName(char* newname){
	strcpy(name, newname);
}

File: Tugas Besar 2/Product.java
/**
 * @file Product.java
 * @author 13515125/Muhammad Rafid Amrullah, 13517092/Michael Ray, 13517113/Juro Sutantra, 13517137/Vincent Budianto
 * @brief file berisi definisi class Product
 * @version 1.0
 * @date 2019-04-21
 */

import java.io.*;
import java.lang.*;
import java.util.*;

/**
 * @brief Product parent class
 * 
 */
public class Product
{
	protected String name;
	protected int price;
	public int prio;

	/**
	 * @brief Create the Farm Product object
	 * 
	 */
	public Product()
	{
		this.price = 0;
		this.name = "product stub";
		this.prio = 0;
	}

	/**
	 * @brief Create the Product object
	 * 
	 */
	public Product(String name, int price, int prio)
	{
		this.price = price;
		this.name = name;
		this.prio = prio;
	}
	
	/**
	 * @brief Get the Name object
	 * 
	 * @return String
	 */
	public String getName()
	{
		return this.name;
	}

	/**
	 * @brief Get the Price object
	 * 
	 * @return int 
	 */
	public int getPrice()
	{
		return this.price;
	}

	/**
	 * @brief Set the Name object
	 * 
	 */
	public void setName(String name)
	{
		this.name = name;
	}
}